pan3d/dataset_builder.py

```diff
@@ -367,11 +430,23 @@ def mesh_changed(self):
 
     def plot_mesh(self):
         self.plotter.clear()
-        self.actor = self.plotter.add_mesh(
-            self.mesh,
+        args = dict(
+            cmap=self.state.render_colormap,
             clim=self.algorithm.data_range,
+            scalar_bar_args=dict(interactive=True),
+        )
+        if self.state.render_transparency:
+            args["opacity"] = self.state.render_transparency_function
+
+        mesh = self.mesh
+        if self.state.render_scalar_warp:
+            mesh = mesh.warp_by_scalar()
+        self.actor = self.plotter.add_mesh(
+            mesh,
+            **args,
         )
         self.plotter.view_isometric()
+        self.plotter.reset_camera()
```

Comment:
Did that part fix the local rendering issue?

Reply:
Sort of; changing the colormap or transparency is a smooth transition now, but changing "warp by scalars" or the x-y-z scales still results in the mesh appearing to move away from the center (or disappear entirely if the change was drastic).

Reply:
So it does not work. Since only the last 2 are what require the camera to be updated.
I don't think it is an API on the plotter that you need to call. This is something on the viewer. And it could be like a push_camera or something. You can use the controller to get your hand on that function.

Reply:
Hm, ok. I have now tried using `self.ctrl.reset_camera()` and `self.ctrl.view_update()`, with different combinations, and I'm able to get the "warp by scalars" change to transistion smoothly with [d2f8baf](https://github.com/Kitware/pan3d/pull/36/commits/d2f8baf773c9a5e68836f96cc45348f1ce20ed0a), but the x-y-z scale changes are behaving the same.

Reply:
You may have to do 


```
self.ctrl.reset_camera() # this line should be the one not needed
self.ctrl.view_update()
self.ctrl.reset_camera()
```

Reply:
Not sure but I think the `view_update` can take a `reset_camera=True` arg

Reply:
It does take that flag, and I'm trying different combinations of the two lines (also switching the order as you suggested), but I am getting the same behavior on these permutations.